usecases/traverser/explorer.go

```diff
@@ -235,18 +237,28 @@ func (e *Explorer) getClassVectorSearch(ctx context.Context,
 func (e *Explorer) searchForTargets(ctx context.Context, params dto.GetParams, targetVectors []string, searchVectorParams []*searchparams.NearVector) ([]search.Result, [][]float32, error) {
 	var err error
 	searchVectors := make([][]float32, len(targetVectors))
+	eg := enterrors.NewErrorGroupWrapper(e.logger)
+	eg.SetLimit(2 * _NUMCPU)
 	for i := range targetVectors {
-		var searchVectorParam *searchparams.NearVector
-		if params.NearVector != nil {
-			searchVectorParam = params.NearVector
-		} else if searchVectorParams != nil {
-			searchVectorParam = searchVectorParams[i]
-		}
+		i := i
+		eg.Go(func() error {
+			var searchVectorParam *searchparams.NearVector
+			if params.NearVector != nil {
+				searchVectorParam = params.NearVector
+			} else if searchVectorParams != nil {
+				searchVectorParam = searchVectorParams[i]
+			}
 
-		searchVectors[i], err = e.vectorFromParamsForTaget(ctx, searchVectorParam, params.NearObject, params.ModuleParams, params.ClassName, params.Tenant, targetVectors[i])
-		if err != nil {
-			return nil, nil, errors.Errorf("explorer: get class: vectorize search vector: %v", err)
-		}
+			searchVectors[i], err = e.vectorFromParamsForTaget(ctx, searchVectorParam, params.NearObject, params.ModuleParams, params.ClassName, params.Tenant, targetVectors[i])
```

Comment:
shouldn't here be mutex used since we change this operation from sequential to parallel?

Reply:
or this is a safe operation since we use index?

Reply:
Not needed - each goroutine only writes into "its" index which is preallocated and there are no reads before the error group finishes